godot-macros/src/class/data_models/inherent_impl.rs

```diff
@@ -51,14 +51,21 @@ pub fn transform_inherent_impl(mut impl_block: venial::Impl) -> ParseResult<Toke
     let (funcs, signals) = process_godot_fns(&class_name, &mut impl_block)?;
     let consts = process_godot_constants(&mut impl_block)?;
 
-    let signal_registrations = make_signal_registrations(signals, &class_name_obj);
+    let signal_registrations = make_signal_registrations(signals.clone(), &class_name_obj);
 
     let method_registrations: Vec<TokenStream> = funcs
+        .clone()
         .into_iter()
         .map(|func_def| make_method_registration(&class_name, func_def))
         .collect::<ParseResult<Vec<TokenStream>>>()?; // <- FIXME transpose this
 
-    let constant_registration = make_constant_registration(consts, &class_name, &class_name_obj)?;
+    let constant_registration =
+        make_constant_registration(consts.clone(), &class_name, &class_name_obj)?;
```

Comment:
do we need to clone here? couldn't this be turned into references instead? it's probably not really a big deal either way.

Reply:
it could

Reply:
it's not really a big deal either way, we can probably come back to this later and make things consistent throughout

Reply:
This can be easily solved by using `make_inherent_impl_docs` take references to all 3 things, and calling it before the other values are consumed. Would be good to do now 😉

Reply:
it still clones inside of make_inherent_impl_docs, i tried it.

Reply:
The current code uses
```rs

    #[cfg(all(feature = "docs", since_api = "4.3"))]
    let docs = crate::docs::make_inherent_impl_docs(&funcs, &consts, &signals);
```
So apparently it worked?

Write this, **don't just resolve the conversation** without awaiting our answer.

Reply:
well it still clones inside of make_inherent_impl_docs, which i dont really think is much better